docs/use_cases/jupyterhub/README.md

```diff
@@ -0,0 +1,147 @@
+It supports both [version 1](https://firecrest.readthedocs.io/en/stable/) and [version 2](https://github.com/eth-cscs/firecrest-v2) of the API.
+It can be deployed on Kubernetes as part of JupyterHub and configured to target different systems.
+
+In this tutorial, we will set up a simplified environment on a local machine, including:
+
+- a [Docker Compose](https://docs.docker.com/compose) deployment of FirecREST, a single-node Slurm cluster and a [Keycloak](https://www.keycloak.org) server which will be used as identity provider for the authentication
+- a local installation of JupyterHub, configured to launch notebooks on the Slurm cluster
+
+This deployment not only demonstrates the use case but also serves as a platform for testing and developing FirecRESTSpawner.
+
+## Requirements
+
+For this tutorial you will need
+
+- a recent installation of Docker, which includes the `docker compose` command (or the older `docker-compose` command line tool)
+- a Python installation (version 3.9 or higher)
+
+## Deployment of FirecREST and Slurm cluster
+
+This tutorial builds on the Docker demo of FirecREST.
+We will use the small [docker-compose-jhub.yaml](./docker-compose-jhub.yaml) file to override some settings in the FirecREST demo.
+This can be done by passing both files to the `docker compose` command.
+
+First, we clone the FirecREST repository
+
+!!! example "Cloning FirecREST official repo"
+    ```bash
+    git clone https://github.com/eth-cscs/firecrest-v2.git
+    ```
+
+and then launch the deployment
+
+!!! example "Launch docker compose"
+    ```bash
+    cd firecrest-v2
+    export JHUB_DOCKERFILE_DIR=./docs/use_cases/jupyterhub/
+    docker compose -f docker-compose.yml -f $JHUB_DOCKERFILE_DIR/docker-compose-jhub.yaml up
+    ```
+
+This step takes a few minutes. In the meanwhile we can install JupyterHub on a local virtual environment.
+
+### Install JupyterHub and FirecRESTSpawner
+
+An easy way to install JupyterHub is via [Miniconda](https://docs.anaconda.com/miniconda/install/).
+We need to [download the Miniconda installer](https://docs.anaconda.com/miniconda/install/) for our platforms and install it using the following command (replace `<arch>` with the architecture on your workstation, for instance `Linux-x86_64` or `MaxOSX-arm64`):
+
+!!! example "Install Miniconda"
+    ```bash
+    bash Miniconda3-latest-<arch>.sh -p /path/to/mc-jhub -b
+    ```
+
+Here we use `-p` to pass the absolute path to the install directory and `-b` to accept the [terms of service](https://legal.anaconda.com/policies/en/).
+
+We can activate our conda base environment and install `configurable-http-proxy`, `JupyterHub` and `FirecRESTSpawner`:
+
+!!! example "Install JupyterHub using FirecRESTSpawner"
+    ```bash
+    . /path/to/mc-jhub/bin/activate
+    conda install -y configurable-http-proxy
+    pip install --no-cache jupyterhub==4.1.6 pyfirecrest==3.0.1 SQLAlchemy==1.4.52 oauthenticator==16.3.1 python-hostlist==1.23.0
```

Comment:
I took the liberty of updating the version of `pyfirecrest` here (in the original it was `2.6.0`)